pkg/controlplane/controller/legacytokentracking/controller.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package legacytokentracking
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+)
+
+const (
+	configMapName    = "tracked-since"
+	configMapDataKey = "timestamp"
+	freezePeriod     = 30 * time.Minute
+)
+
+// Controller maintains a bool value configmap `tracked-since` in `kube-system`
+// to indicates if the tracking for legacy tokens is enabled in the cluster.
+// For HA clusters, the configmap will be eventually created after all
+// controller instances have enabled the feature. When disabling this feature,
+// existing configmap will be deleted.
+type Controller struct {
```

Comment:
I know it's namespaced but I still prefer having full names instead of generic ones that could appear multiple times around the codebase and make searches harder.